src/lightning_lite/lite.py

```diff
@@ -159,19 +166,77 @@ def setup(
             model = self._move_model_to_device(model=model, optimizers=list(optimizers))
 
         # Let accelerator/plugin wrap and connect the models and optimizers
-        model, optimizers = self._strategy.setup_module_and_optimizers(model, list(optimizers))
+        if optimizers:
+            model, optimizers = self._strategy.setup_module_and_optimizers(  # type: ignore[assignment]
+                model, list(optimizers)
+            )
+        else:
+            model = self._strategy.setup_module(model)
+
         model = _LiteModule(model, self._precision, original_module=original_model)
 
         # Update the _DeviceDtypeModuleMixin's device parameter
         model.to(self.device if move_to_device else next(model.parameters()).device)
 
         optimizers = [_LiteOptimizer(optimizer=optimizer, strategy=self._strategy) for optimizer in optimizers]
+
         self._models_setup += 1
+
         if optimizers:
-            # join both types in a list for API convenience
-            return [model] + optimizers
+            # join both types in a tuple for API convenience
+            return tuple((model, *optimizers))
+        return model
+
+    def setup_model(self, model: nn.Module, move_to_device: bool = True) -> _LiteModule:
```

Comment:
Since we are introducing these specific blocks, I think we should rename `def setup()` to `def setup_module_and_optimizers` to avoid confusion from users who don't know which one to use

Reply:
No, that's too long. Just setup is ok. Since it does not specify anything in the name, the intuition is that it can setup multiple things which it does. And error messaging will guide the user eventually.

Reply:
@lantiga Can you clarify whether you meant that:

A) `setup()` kept as-is (polymorphic). We introduce `setup_model` and `setup_optimizer` for the cases where one needs to initialize them separately.
B) `setup` renamed to `setup_model_and_optimizer` (explicit) where both have to be passed. We introduce `setup_model` and `setup_optimizer` for the cases where one needs to initialize them separately.

This is in reference to your comment:

> I don’t like the polymorphic setup method. I think we should be explicit:
setup will work for all strategies that don’t need to wrap the submodules.

PR currently implements A). My question is if we should remove all polymorphism and do B)

Reply:
A) should be the preferred way IMO, as this covers the vast majority of use cases. The name `setup_model_and_optimizer` is too long and not as easy to remember as just setup().

Reply:
I agree with @awaelchli here; it is just way more convenient to call `.setup`